### DISCUSSION TITLE
Which algorithms do you want to see implemented?

### DISCUSSION THREAD

**Discussion by SkepticMystic** [SkepticMystic/graph-analysis#23] [General]
We are open to adding new algorithms to analyse the vault :) Please share which one's you'd like to see implemented, and ideally post a concrete algorithm we can use (like the formulas posted in the readme).

Here is where most of the existing algs came from: [Neo4j Docs](https://neo4j.com/docs/graph-data-science/current/algorithms/)

**Comment by dleeftink** [comment]
A minimum spanning tree or breadth first search implementation would be a useful addition to extract a graph 'backbone'. BFS for instance, could be used to generate different layers of degree from the current note (e.g. first level of degree, second level, etc.). MST in return, yields a topological list of edges or a "path of least resistance" between all notes.

**Reply by SkepticMystic** [reply]
That is useful information, indeed.
I am unsure of how we would present that, though. What are your thoughts on this?

**Reply by dleeftink** [reply]
I think BFS can best be represented as a (collapsible) [indented tree](https://observablehq.com/@d3/indented-tree), like so:

[[current note]]
- [[first degree note A]]
  - [[second degree note C]]
  - [[second degree note E]]
    - [[third degree note F]]
      - [[fourth degree note G]]
      - [ ... ]
- [[first degree note B]]
  - [[second degree note D]]
  - [ ... ]

This has the added benefit of clearly organising closely linked notes and leaving room for graph statistics such as degree, betweenness, etc.

___

Representing an MST can either be done via above method (barring cycles and notes with multiple parents), a topologically sorted list of edges, or if Obsidian implements and edge filtering function, the addition of a [ backbone :: outgoing link ] property (YAML/view) field to notes.

As edge filtering isn't possible currently, what may be an interim solution is to write the results to a flat folder of notes named MST/{{result name}} and populate it with notes only containing outgoing links as calculated by MST as well as a link/embed to the original note. This way, users can filter the graph to only display the MST/{{result name}} folder and get a sense of the graph backbone (folder population might be limited to a user set degree, e.g. returning notes that have at least n-degree).

The latter approach is less than ideal, but the only way besides implementing a custom graph view (i.e. Juggl). However, this method of generating a duplicate folder of notes paves way for subgraph extraction and creating a 'safe' copy of the main graph that can be enriched with all sorts of analytical fields, while still preserving the link to the original note via an embed. Think of a derivative graph that can be freely manipulated by this plugin in an environment separate from the main graph.
